airflow/providers/databricks/sensors/databricks.py

```diff
@@ -0,0 +1,231 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""This module contains Databricks sensors."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any, Callable, Sequence
+
+from airflow.exceptions import AirflowException
+from airflow.providers.common.sql.hooks.sql import fetch_all_handler
+from airflow.providers.databricks.hooks.databricks_sql import DatabricksSqlHook
+from airflow.sensors.base import BaseSensorOperator
+from airflow.utils.context import Context
+
+
+class DatabricksSqlSensor(BaseSensorOperator):
+    """
+    Generic Databricks SQL sensor.
```

Comment:
I would make it really generic by allowing to pass an arbitrary SQL expression that will trigger sensor if it has non-empty result. And then on top of it we can build Partition & History change sensors.

Reply:
@alexott Sure, I can make the change. One clarification - the idea is:

1. Creating a generic sensor class with SQL expression input. 
2. Create child classes of it for partition and history changes with additional condition checks in the child classes.

Does that sound like a solid design to you for extensibility to other sensor types in future?

Reply:
There are few use cases for generic SQL support:

* People want to detect when new data is available, for example, for today (it will be dynamic every day). And timestamp may not always be partition, especially for Delta with its data skipping functionality that making dependency on partitioning less needed.
* Partitions sensor is a specific case of SQL sensor - basically we just need to generate a correct SQL & pass to generic SQL sensor.
* Table changes may have a completely different implementation because it needs to store a "latest state" somewhere, but that's ok. And it's still can reuse pieces of the generic SQL sensor implementation.

Reply:
Thank you, good points keeping the SQL sensor generic. I am working on modifying the branch accordingly. I will push the changes soon.

Reply:
Implemented the changes and just pushed them to the PR.